libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/HttpConfigurationExtensions.cs

```diff
@@ -29,73 +29,63 @@ public static HttpConfiguration MapBotFramework(this HttpConfiguration httpConfi
 
             configurer?.Invoke(optionsBuilder);
 
-            var botFrameworkAdapter = GetOrCreateBotFrameworkAdapter();
+            var botFrameworkAdapter = httpConfiguration.DependencyResolver.GetService(typeof(BotFrameworkAdapter)) as BotFrameworkAdapter;
+            if (botFrameworkAdapter == null)
+            {
+                var credentialProvider = ResolveCredentialProvider(options);
 
-            ConfigureMiddleware(botFrameworkAdapter);
-            ConfigureBotRoutes(botFrameworkAdapter);
+                // TODO: fix up constructor to take options
+                botFrameworkAdapter = new BotFrameworkAdapter(credentialProvider, options.ConnectorClientRetryPolicy, options.HttpClient);
+            }
 
-            return httpConfiguration;
+            // error handler
+            botFrameworkAdapter.ErrorHandler = options.ErrorHandler;
 
-            BotFrameworkAdapter GetOrCreateBotFrameworkAdapter()
+            // add middleware
+            foreach (var middleware in options.Middleware)
             {
-                if (!(httpConfiguration.DependencyResolver.GetService(typeof(BotFrameworkAdapter)) is BotFrameworkAdapter adapter))
-                {
-                    var credentialProvider = ResolveCredentialProvider();
+                botFrameworkAdapter.Use(middleware);
+            }
 
-                    adapter = new BotFrameworkAdapter(credentialProvider, options.ConnectorClientRetryPolicy, options.HttpClient);
-                }
+            ConfigureBotRoutes(httpConfiguration, options, botFrameworkAdapter);
 
-                return adapter;
-            }
+            return httpConfiguration;
+        }
 
-            void ConfigureMiddleware(BotFrameworkAdapter adapter)
-            {
-                foreach (var middleware in options.Middleware)
-                {
-                    adapter.Use(middleware);
-                }
-            }
+        private static void ConfigureBotRoutes(HttpConfiguration httpConfiguration, BotFrameworkOptions options, BotFrameworkAdapter adapter)
```

Comment:
I honestly see no justification for this to have been changed from a local function; so much more noise created. If the team is taking a stance that local functions are verboten then the project needs to set up a `CONTRIBUTING.md` that spells out such rules, otherwise this feels like a change due to personal preference that should have not have been made given the scope of the actual work being done here.

Even just in terms of code churn, it created more for people to review... doesn't seem to have been beneficial in any way.